CHEWBBACA/CHEWBBACA_NS/down_profiles.py

```diff
@@ -0,0 +1,329 @@
+		print("\nfetching metadata...\n")
+		pool = multiprocessing.Pool(cpu2use)
+		result2 = Result()
+
+		for uri in orderedkeys:
+			p = pool.apply_async(get_isol_metadata, args=[uri], callback=result2.update_result)
+
+		pool.close()
+		pool.join()
+		listResults2= result2.get_result()
+	else:
+		print("no genomes to download or fetch metadata")
+		return
+
+	#get all possible metadata headers
+	allheaders=[]
+	for isolate in listResults2:
+		for headers in isolate.keys():
+			allheaders.append(headers)
+
+	allheaders=sorted(set(allheaders))
+
+	print("\nwriting metadata\n")
+	if len(allheaders)>0:
+		with open("metadata.tsv", "w") as f:
+			allheaders.remove('FILE')
+			f.write('FILE\t')
+			f.write(('\t'.join(map(str, allheaders))) + "\n")
+
+			for isolate in listResults2:
+				towrite=[isolate['FILE']]
+				for header in allheaders:
+					try:
+						towrite.append(isolate[header])
+					except:
+						towrite.append('NA')
+				f.write(('\t'.join(map(str, towrite))) + "\n")
+
+		print("Metadata is now available at metadata.tsv")
+
+	#create final results list of profiles
+	listFinalProfile=[["FILE"]+orderedLoci]
+
+	for isolate in listResults:
+		aux=[]
+		isolUri=isolate[0]
+		profile=isolate[1]
+		#aux.append(dictIsol[isolUri])
+		aux.append(isolUri)
+		for loci in orderedLoci:
+			alleleId=0
+			try:
+				alleleId=(profile[loci])[0]
+			except:
+				pass
+			aux.append(alleleId)
+
+		listFinalProfile.append(aux)
+
+
+	if len(dictIsol.keys()) == 0:
+		print("no profiles were downloaded")
+
+	elif equal_headers:
+		listFinalProfile.pop(0)
+		toWrite=''
+
+		with open(inputProfile) as f1:
+			for line in f1:
+				toWrite+=line
+		with open("profiles.tsv", "w") as f:
+				#f.write(line)
+			f.write(toWrite)
+			for profile in listFinalProfile:
+				# ~ print (profile)
+				f.write(('\t'.join(map(str, profile))) + "\n")
+		print("Profile is now available at profiles.tsv")
+
+	elif len(dictIsol.keys()) >0:
+		with open("profiles.tsv", "w") as f:
+			for profile in listFinalProfile:
+				#~ print (profile)
+				f.write(('\t'.join(map(str, profile)))+"\n")
+		print("Profile is now available at profiles.tsv")
+
+	#with open("keys.tsv","w") as f:
+	#	f.write("URI\tName"+"\n")
+	#	for uri in dictIsol.keys():
+	#		f.write(uri+"\t"+dictIsol[uri]+"\n")
+
+def main(species,schema,cpu2use,inputProfile,listGenomes=None,token=None):
+
+	server_url=(species.split("/species"))[0]
+
+	r = requests.get(server_url, timeout=5)
+	if r.status_code >200:
+		print("server returned error with code "+str(r.status_code)+", program will stop since there seems to be a problem contacting the server")
+		return
+	else:
+		print("server is running, will start the program")
+
+	#build list of genomes to download if provided, if not it downloads all profiles which take a while
+	try:
+		with open(listGenomes) as f:
+			content = f.readlines()
+		genomes2Down = [x.strip() for x in content]
+
+	except:
+		genomes2Down=[]
+		pass
+
+	print(inputProfile)
+	print("laa")
```

Comment:
HAHAHA! ;)

Reply:
This file is still in development and was removed from the branch to avoid merging unused code into Master. But we did removed the `laa` of despair :)